### ISSUE TITLE
Decimal scientific notation json number fields are not parsed by vm.parseJson

### ISSUE THREAD

**Issue by gte620v** [foundry-rs/foundry#6006]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

87283bc

### What command(s) is the bug in?

forge test

### Operating System

macOS (Apple Silicon)

### Describe the bug

See https://github.com/foundry-rs/foundry/issues/5038#issuecomment-1751727791 and https://github.com/foundry-rs/foundry/pull/5928#issuecomment-1751337269

It seems that this [commit](https://github.com/foundry-rs/foundry/pull/5928) broke this test case:

```solidity
  function test_parse_11e20_sci() public {
    string memory json = '{"value": 1.1e20}';
    bytes memory parsed = vm.parseJson(json);
    Value memory data = abi.decode(parsed, (Value));
    assertEq(data.value, 1.1e20);
  }
```

That passes as recently as this commit [7b291d4](https://github.com/foundry-rs/foundry/commit/7b291d43778be31620a9b45b19a68339f7286f08)

```
(returns [PASS] test_parse_11e20_sci() (gas: 4124))
```

On `87283bc`, this test returns:

```
[FAIL. Reason: Failed to parse key "$": Unsupported value: Number(1.1e20)] test_parse_11e20_sci() (gas: 3308)
```

cc @mds1

**Comment by mattsse** [comment]
thanks for flagging.

likely a regression of 

https://github.com/foundry-rs/foundry/pull/5882

**Comment by mattsse** [comment]
fixed in #6009 

thanks for finding the right commit where the invalid check for scientific notation was introduced.
